src/html5_parser/node.rs

```diff
@@ -592,167 +592,136 @@ mod test {
     }
 
     #[test]
-    fn test_type_of_node_data_document() {
+    fn type_of_node_data_document() {
         let node = Node::new_document();
         assert_eq!(node.data, NodeData::Document {});
     }
 
     #[test]
-    fn test_contains_attribute_non_element() {
+    fn contains_attribute_non_element() {
         let node = Node::new_document();
-        let result = node.contains_attribute("x".to_string());
+        let result = node.contains_attribute("x");
         assert!(result.is_err())
     }
 
     #[test]
-    fn test_contains_attribute() {
+    fn contains_attribute() {
         let mut attr = HashMap::new();
         attr.insert("x".to_string(), "value".to_string());
 
         let node = Node::new_element("node", attr.clone(), HTML_NAMESPACE);
 
-        match node.contains_attribute("x".to_string()) {
-            Err(_) => assert!(false),
-            Ok(result) => {
-                assert_eq!(result, true);
-            }
-        }
-
-        match node.contains_attribute("z".to_string()) {
-            Err(_) => assert!(false),
-            Ok(result) => {
-                assert_eq!(result, false);
-            }
-        }
+        assert!(node.contains_attribute("x").unwrap());
+        assert!(!node.contains_attribute("z").unwrap());
     }
 
     #[test]
-    fn test_insert_attrubte_non_element() {
+    fn insert_attrubte_non_element() {
         let mut node = Node::new_document();
-        let result = node.insert_attribute("name".to_string(), "value".to_string());
+        let result = node.insert_attribute("name", "value");
         assert!(result.is_err());
     }
 
     #[test]
-    fn test_insert_attribute() {
+    fn insert_attribute() {
         let attr = HashMap::new();
         let mut node = Node::new_element("name", attr.clone(), HTML_NAMESPACE);
-        if let Ok(_) = node.insert_attribute("key".to_string(), "value".to_string()) {
-            if let Ok(value) = node.get_attribute("key".to_string()) {
-                match value {
-                    None => assert!(false),
-                    Some(val) => assert_eq!(*val, "value".to_string()),
-                }
-            }
-        }
+
+        assert!(node.insert_attribute("key", "value").is_ok());
+        let value = node.get_attribute("key").unwrap().unwrap();
+        assert_eq!(value, "value");
```

Comment:
I had a question about whether a `Result` is needed here, since we're already using an `Option`.